src/data/polynomial/degree/trailing_degree.lean

```diff
@@ -189,27 +194,44 @@ by simpa only [C_1, one_mul] using le_trailing_degree_C_mul_X_pow (1:R) n
 theorem le_trailing_degree_X : (1 : with_top ℕ) ≤ trailing_degree (X : polynomial R) :=
 by simpa only [C_1, one_mul, pow_one] using le_trailing_degree_C_mul_X_pow (1:R) 1
 
+lemma support_X (H : (1:R) ≠ 0) : (X : polynomial R).support = singleton 1 :=
```

Comment:
These two lemmas should move to another file. Probably `../basic.lean`

Reply:
Sounds good: I'll add them to basic.lean and remove them from here!

Reply:
I pulled the latest version of mathlib, before moving around support_X, but now VSC hangs for a long time trying to process the files data/polynomial/degree/basic.lean or data/polynomial/degree/trailing_degree-lean and eventually crashes.  Is this an issue on my computer or is it something that will naturally solve itself, by waiting for some time?

Reply:
> I pulled the latest version of mathlib, before moving around support_X, but now VSC hangs for a long time trying to process the files data/polynomial/degree/basic.lean or data/polynomial/degree/trailing_degree-lean and eventually crashes. Is this an issue on my computer or is it something that will naturally solve itself, by waiting for some time?

You can tryp "leanproject get-cache" (to pull pre-compiled stuff from github), or "leanproject build" + a 30-minute break (to compile the whole thing locally).

Reply:
Thank you for the suggestion: "leanproject get-cache" worked great!

Reply:
@jcommelin Dear Johan, I think that I moved the support_X lemmas to basic, but I am not sure that the files got updated on GitHub, after I pushed them.  I am pretty sure that trailing_degree.lean has been updated, but not so sure about basic.lean